### ISSUE TITLE
HSTS + Mailhog

### ISSUE THREAD

**Issue by nathanielks** [roots/trellis#689]
## Submit a feature request or bug report

- [X] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/guidelines/blob/master/CONTRIBUTING.md)
- [ ] This is a feature request
- [X] This is a bug report
- [ ] This request isn't a duplicate of an [existing issue](https://github.com/roots/trellis/issues)
- [ ] I've read the [docs](https://roots.io/trellis/docs) and followed them (if applicable)

---

(I'm just doing this so I remember to investigate further)

**What is the current behavior?**

HTTPS in development with HSTS enabled does not work with Mailhog because of conflicting ports.

**What is the expected or desired behavior?**

IT WORK :trollface:

**Comment by retlehs** [comment]
not able to reproduce on a dev site with SSL + HSTS enabled - was able to go to mailhog without https

closing out as this is a year old and a lot of moving parts have changed since this report

**Comment by jasonmarlin** [comment]
To piggyback OP, this isn't really a port conflict, but that browsers want to redirect to HTTPS on an entire domain if HSTS is enabled. I propose that development environment could include a mailhog proxy in Nginx to ease both access to mailhog and eliminate any HSTS issues with HTTPS redirection. The following code added to `wordpress-site.conf.j2` does the trick. Happy to add a PR:

```
  {% if env == 'development' -%}
  {% block mailhog_proxy -%}
  location /mailhog/ {
    proxy_pass http://127.0.0.1:8025/;
    proxy_set_header Host $host;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
  }
  {%- endblock %}
  {% endif -%}
```

**Comment by swalkinshaw** [comment]
@jasonmarlin that solution is interesting. I'd welcome a PR but is this an actual issue? @retlehs couldn't reproduce it.

You had the same issue?

**Comment by jasonmarlin** [comment]
I did have the same issue and have seen it in other projects. HSTS spec
wants to redirect any http:// request to https:// across a domain,
regardless of the port number. That said, my proposed fix via Nginx proxy
isn't 100% as Nginx doesn't seem to know how to upgrade the web socket
requests to wss, and I ran out of time to research.

Another option could be to setup the mailhog server to work over https/wss,
but I couldn't find any quick solution for that either.

On Fri, Jan 19, 2018 at 11:27 AM, Scott Walkinshaw <notifications@github.com
